server/valgrind-out.txt

```diff
@@ -1,113 +1,134 @@
+--20658-- REDIR: 0x491e100 (libc.so.6:index) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491e630 (libc.so.6:strlen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x4927bb0 (libc.so.6:memrchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491f850 (libc.so.6:strcasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491f3d0 (libc.so.6:memchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x493b930 (libc.so.6:wcslen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491eae0 (libc.so.6:strspn) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491f750 (libc.so.6:stpncpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491f6f0 (libc.so.6:stpcpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x4920c90 (libc.so.6:strchrnul) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x491f8a0 (libc.so.6:strncasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--20658-- REDIR: 0x49308c0 (libc.so.6:__strrchr_sse2) redirected to 0x483ea70 (__strrchr_sse2)
+--20658-- REDIR: 0x493af00 (libc.so.6:__memset_sse2_unaligned) redirected to 0x48428e0 (memset)
+--20658-- REDIR: 0x49190e0 (libc.so.6:malloc) redirected to 0x483b780 (malloc)
+--20658-- REDIR: 0x49196d0 (libc.so.6:free) redirected to 0x483c9d0 (free)
+--20658-- REDIR: 0x491ab10 (libc.so.6:calloc) redirected to 0x483dce0 (calloc)
+--20658-- REDIR: 0x49306b0 (libc.so.6:__strchrnul_sse2) redirected to 0x4843540 (strchrnul)
+--20658-- REDIR: 0x4930bc0 (libc.so.6:__strlen_sse2) redirected to 0x483efa0 (__strlen_sse2)
+--20658-- REDIR: 0x493aad0 (libc.so.6:memcpy@GLIBC_2.2.5) redirected to 0x4840100 (memcpy@GLIBC_2.2.5)
+--20658-- REDIR: 0x491efa0 (libc.so.6:__GI_strstr) redirected to 0x4843ca0 (__strstr_sse2)
+--20658-- REDIR: 0x4a0c790 (libc.so.6:__memcmp_sse4_1) redirected to 0x4842150 (__memcmp_sse4_1)
+--20658-- REDIR: 0x49293b0 (libc.so.6:__strcmp_sse2_unaligned) redirected to 0x483fed0 (strcmp)
+--20658-- REDIR: 0x493aab0 (libc.so.6:__mempcpy_sse2_unaligned) redirected to 0x4843660 (mempcpy)
+--20658-- REDIR: 0x4919e80 (libc.so.6:realloc) redirected to 0x483df30 (realloc)
+==20658== Conditional jump or move depends on uninitialised value(s)
+==20658==    at 0x10B8AE: main (in /home/vsrlinux/Desktop/assignments/a3/assignments-3-and-later-SrirajVemparala/server/aesdsocket)
+==20658==  Uninitialised value was created by a heap allocation
+==20658==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==20658==    by 0x10B67E: main (in /home/vsrlinux/Desktop/assignments/a3/assignments-3-and-later-SrirajVemparala/server/aesdsocket)
+==20658== 
+==20658== Warning: invalid file descriptor -1 in syscall close()
+==20658==    at 0x486F45B: close (close.c:27)
+==20658==    by 0x10BA6E: main (in /home/vsrlinux/Desktop/assignments/a3/assignments-3-and-later-SrirajVemparala/server/aesdsocket)
```

Comment:
@SrirajVemparala these are all most likely real issues, Add the `-g` option to compile flags to have it turn on debug info and give you line numbers.